spec/models/reservation_spec.rb

```diff
@@ -26,4 +26,38 @@
   it 'validates raise error for invalid status' do
     expect { create(:reservation, status: 'invalid') }.to raise_error(ArgumentError, "'invalid' is not a valid status")
   end
+
+  describe 'associations' do
+    it "ensure to don't create a reservation with an unavailable room" do
+      _available_room = create(:available_double_room)
+      unavailable_room = create(:unavailable_double_room)
+      reservation = create(:reservation, room_id: unavailable_room.id)
+
+      expect(reservation.room.status).to eq('available')
+    end
+
+    it 'validates to assign a guest' do
+      guest = create(:guest)
+      reservation = create(:reservation, guest_id: guest.id)
+
+      expect(reservation.guest).to eq(guest)
+    end
```

Comment:
Not sure there is a lot of point to test this. I think what you want is to test association, not validation. You don't really need to test associations. But if you really want, you could use [shoulda gem](https://github.com/thoughtbot/shoulda) to make your life easier.